File: src/com/darin/weex/weexToolKit/WeexProcess.java
package com.darin.weex.weexToolKit;

/**
 * Created by darin on 7/14/16.
 */
public class WeexProcess {
    public Process getProcess() {
        return process;
    }

    public long getPreviewServerPort() {
        return previewServerPort;
    }

    public long getWebServicePort() {
        return webServicePort;
    }

    private Process process;
    private long previewServerPort;
    private long webServicePort;

    public String getWeexFilePath() {
        return weexFilePath;
    }

    private String weexFilePath;

    private WeexProcess(Builder builder) {
        process = builder.getProcess();
        previewServerPort = builder.getPreviewServerPort();
        webServicePort = builder.getWebServicePort();
        weexFilePath = builder.getWeexFileName();
    }


    public boolean destory() {
        if (process == null)
            return true;

        process.destroy();
        try {
            process.exitValue();
        } catch (IllegalThreadStateException e) {
            return false;
        }

        process = null;
        weexFilePath = null;

        return true;
    }

    static class Builder {
        private Process process;

        public long getPreviewServerPort() {
            return previewServerPort;
        }

        public Process getProcess() {
            return process;
        }

        public long getWebServicePort() {
            return webServicePort;
        }

        private long previewServerPort = 8081;
        private long webServicePort = 8082;

        public String getWeexFileName() {
            return weexFileName;
        }

        private String weexFileName;

        public Builder(Process process) {
            this.process = process;
        }

        public Builder previewServerPort(long port) {
            previewServerPort = port;
            return this;
        }

        public Builder weexFileName(String name) {
            weexFileName = name;
            return this;
        }

        public Builder webServicePort(long port) {
            webServicePort = port;
            return this;
        }

        public WeexProcess build() {
            return new WeexProcess(this);
        }


    }
}

File: src/com/darin/weex/weexToolKit/WeexProcess.kt
package com.darin.weex.weexToolKit

/**
 * Created by darin on 7/14/16.
 */
class WeexProcess private constructor(builder: WeexProcess.Builder) {

    var process: Process? = null
        private set
    val previewServerPort: Int
    val webServicePort: Int

    var weexFilePath: String? = null
        private set

    init {
        process = builder.process
        previewServerPort = builder.previewServerPort
        webServicePort = builder.webServicePort
        weexFilePath = builder.weexFileName
    }


    fun destory(): Boolean {
        if (process == null)
            return true

        process!!.destroy()
        try {
            process!!.exitValue()
        } catch (e: IllegalThreadStateException) {
            return false
        }

        process = null
        weexFilePath = null

        return true
    }

    internal class Builder(val process: Process) {

        var previewServerPort: Int = 8081
            private set
        var webServicePort: Int = 8082
            private set

        var weexFileName: String? = null
            private set

        fun previewServerPort(port: Int): Builder {
            previewServerPort = port
            return this
        }

        fun weexFileName(name: String): Builder {
            weexFileName = name
            return this
        }

        fun webServicePort(port: Int): Builder {
            webServicePort = port
            return this
        }

        fun build(): WeexProcess {
            return WeexProcess(this)
        }


    }
}
